### DISCUSSION TITLE
"FIPS mode not supported" on OpenSSL 1.1.1

### DISCUSSION THREAD

**Discussion by dasariprasad1** [openssl/openssl#21445] [Q&A] [answered]
I see "FIPS mode is not supported" on openssl version 1.1.1. We have requirement only to work with openssl 1.1.1 and looking for option to enable fips in 1.1.1. 
I come across this link https://beta.openssl.org/docs/fips/UserGuide-1.1.1.pdf for generating fips object module that works with openssl 1.1.1. But the specific distribution link mentioned in this document is not working http://www.openssl.org/source/openssl-fips-1.1.1.tar.gz. 

Any suggestions on enabling fips with openssl 1.1.1 version.

**Comment by paulidale** [comment, marked as answer]
There is no FIPS mode for OpenSSL 1.1.1 and never will be.

If you want FIPS support, you must use OpenSSL 3.x.

**Reply by t-j-h** [reply]
Note that there is no FIPS140 validated solution **from the project**.
There are various solutions from other downstream vendors.

**Reply by dasariprasad1** [reply]
I see link is not updated, I am adding here again http://www.openssl.org/source/openssl-fips-1.1.1.tar.gz. Can you help in getting this file.

**Reply by t8m** [reply]
There is no such file.

**Comment by mattcaswell** [comment]
> I come across this link https://beta.openssl.org/docs/fips/UserGuide-1.1.1.pdf for generating fips object module that works with openssl 1.1.1

The "1.1.1" referenced there is *not* referring to OpenSSL 1.1.1. It's the FIPS module version. The date on that document is 2007. The first release of OpenSSL 1.1.1 was in 2018. Don't use that module. You seem to have found an old out of date document on our beta site that no longer exists on the "real" website.

As noted above 1.1.1 has not been and never will be validated by the project. You might find a module from another vendor but we cannot advise you on that.

**Comment by dasariprasad1** [comment]
Ok, Thank you for the update..
